src/main/java/com/amazon/opendistroforelasticsearch/security/resolver/IndexResolverReplacer.java

```diff
@@ -351,53 +333,61 @@ public Resolved resolveRequest(final Object request) {
 
     public final static class Resolved implements Serializable, Writeable {
 
-        /**
-         *
-         */
-        private static final Set<String> All_SET = Collections.singleton("*");
+        private static final Set<String> All_SET = ImmutableSet.of("*");
+        private static final Set<String> types = All_SET;
         private static final long serialVersionUID = 1L;
-        public final static Resolved _LOCAL_ALL = new Resolved(All_SET, All_SET, All_SET, Collections.emptySet());
+        public static final Resolved _LOCAL_ALL = new Resolved(All_SET, All_SET, All_SET, Collections.emptySet());
+
         private final Set<String> aliases;
         private final Set<String> allIndices;
-        private static final Set<String> types = ImmutableSet.of("*");
-
         private final Set<String> originalRequested;
         private final Set<String> remoteIndices;
+        private final boolean isLocalAll;
 
-        private Resolved(final Set<String> aliases, final Set<String> allIndices,
-                         final Set<String> originalRequested, final Set<String> remoteIndices) {
-            this.aliases = aliases;
-            this.allIndices = allIndices;
-            this.originalRequested = originalRequested;
-            this.remoteIndices = remoteIndices;
+        private Resolved(final Collection<String> aliases,
+                         final Collection<String> allIndices,
+                         final Collection<String> originalRequested,
+                         final Collection<String> remoteIndices) {
+            this.aliases = ImmutableSet.copyOf(aliases);
+            this.allIndices = ImmutableSet.copyOf(allIndices);
+            this.originalRequested = ImmutableSet.copyOf(originalRequested);
+            this.remoteIndices = ImmutableSet.copyOf(remoteIndices);
+            this.isLocalAll = IndexResolverReplacer.isLocalAll(originalRequested) || (aliases.contains("*") && allIndices.contains("*"));
         }
 
-        public boolean isLocalAll() {
-            if(IndexResolverReplacer.isLocalAll(originalRequested==null?null:originalRequested.toArray(new String[0]))) {
-                return true;
-            }
+        public Resolved(final StreamInput in) throws IOException {
+            this(in.readList(StreamInput::readString),
+                    in.readList(StreamInput::readString),
+                    in.readList(StreamInput::readString),
+                    in.readList(StreamInput::readString));
+        }
 
-            return aliases.contains("*") && allIndices.contains("*") && types.contains("*");
```

Comment:
Removed `types.contains("*")` check as it always is true as types is defined as `ImmutableSet.of("*");`